proto/src/determined/trial/v1/trial.proto

```diff
@@ -103,3 +103,50 @@ message Trial {
   // Best checkpoint.
   CheckpointWorkload best_checkpoint = 10;
 }
+
+// TrialProfilerMetricLabels are the labels for a single series, where a series
+// is a defined as all metrics sharing a distinct set of labels
+message TrialProfilerMetricLabels {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "trial_id", "name" ] }
+  };
+  // To distinguish the 2 different categories of metrics.
+  enum ProfilerMetricType {
+    // Zero-value (not allowed).
+    PROFILER_METRIC_TYPE_UNSPECIFIED = 0;
+    // For systems metrics, like GPU utilization or memory.
+    PROFILER_METRIC_TYPE_SYSTEM = 1;
+    // For timing metrics, like how long a backwards pass or getting a batch
+    // from the dataloader took.
+    PROFILER_METRIC_TYPE_TIMING = 2;
+  }
+  // The ID of the trial.
+  int32 trial_id = 1;
+  // The name of the metric.
+  string name = 2;
+  // The agent ID associated with the metric.
+  string agent_id = 3;
+  // The GPU UUID associated with the metric.
+  string gpu_uuid = 4;
+  // The type of the metric.
+  ProfilerMetricType metric_type = 5;
+}
+
+// TrialProfilerMetricsBatch is a batch of trial profiler metrics. A batch will
+// contain metrics pertaining to a single series. The fields values, batches and
+// timestamps will be equal length arrays with each index corresponding to a
+// reading.
+message TrialProfilerMetricsBatch {
+  option (grpc.gateway.protoc_gen_swagger.options.openapiv2_schema) = {
+    json_schema: { required: [ "values", "batches", "timestamps", "labels" ] }
+  };
+  // The measurement for a reading, repeated for the batch of metrics.
+  repeated float values = 1;
```

Comment:
Why did we choose this format of three independent lists instead of a list of Metric objects where each Metric object holds `(value, batch, timestamp)`?

Reply:
nothing against that, i guess this is just exactly how it's stored. @hkang1 thoughts?

Reply:
WebUI can take advantage of a list of objects to only traverse all of the data once instead of 4x (once for each field). 🙏

Reply:
actually.. i really don't want to do this. if we wrap these, they're all going to need to be in an object, so instead of
```
{ batches: [ ... ], values: [ ... ], timestamps: [ ... ] }
```
We end up with 
```
{ metrics: [ { batch: X, value: X, timestamp: X }, { ... }] }
```
which is a lot more overhead when we're using the grpc-gateway APIs without any compression.

I know I've been pretty picky about the kinda stuff, I just don't want to get pushed too far away from what has been benchmarked again.

Reply:
the original approach works for the WebUI too. We just need to know that `batches`, `values`, and `timestamps` will all return the same sized lists.